libs/category/src/effects/category.effects.ts

```diff
@@ -143,18 +150,70 @@ export class DaffCategoryEffects {
 	)
 	
 	private toggleCategoryFilter(
-		toggledFilter: DaffCategoryFilterAction, 
+		toggledFilter: DaffToggledCategoryFilterAction, 
 		appliedFilters: DaffCategoryFilterAction[]
 	): DaffCategoryFilterAction[] {
-		return appliedFilters.find(filter => this.areFiltersEqual(toggledFilter, filter))
+		return this.isFilterApplied(toggledFilter, appliedFilters)
 		// filter is applied, remove it
-		? appliedFilters.filter(filter => !this.areFiltersEqual(filter, toggledFilter))
+		? this.removeFilter(toggledFilter, appliedFilters)
 		// filter is not applied, add it
-		: appliedFilters.concat(toggledFilter);
+		: this.addFilter(toggledFilter, appliedFilters)
+	}
+
+	private isFilterApplied(toggledFilter: DaffToggledCategoryFilterAction, appliedFilters: DaffCategoryFilterAction[]): boolean {
+		return !!appliedFilters.find(filter => 
+			toggledFilter.name === filter.name && !!filter.values.filter(value => value === toggledFilter.value).length
+		);
 	}
 
-	private areFiltersEqual(filter1: DaffCategoryFilterAction, filter2: DaffCategoryFilterAction): boolean {
-		return filter1.name === filter2.name && filter1.value === filter2.value;
+	private removeFilter(toggledFilter: DaffToggledCategoryFilterAction, appliedFilters: DaffCategoryFilterAction[])
+		: DaffCategoryFilterAction[] {
+		return appliedFilters.map(filter => {
+			if(filter.name === toggledFilter.name) {
+				filter.values = filter.values.filter(value => value !== toggledFilter.value);
+			}
+
+			return filter;
+		}).filter(filter => !!filter.values.length);
+	}
+
+	private addFilter(toggledFilter: DaffToggledCategoryFilterAction, appliedFilters: DaffCategoryFilterAction[])
+		: DaffCategoryFilterAction[] {
+		if(appliedFilters && appliedFilters.filter(filter => filter.name === toggledFilter.name).length) {
+			return appliedFilters.map(filter => {
+				if(filter.name === toggledFilter.name) {
+					filter.values.push(toggledFilter.value);
+				}
+
+				return filter;
+			})
+		}
+
+		return appliedFilters.concat([{
+			name: toggledFilter.name,
+			label: toggledFilter.label,
+			values: [toggledFilter.value],
+			type: toggledFilter.type
+		}]);
+	}
+	
+	validateFilters(filters: DaffCategoryFilterAction[]): void {
```

Comment:
This validation should be done as much as possible at compile time instead of runtime. I've left a comment on the model to attempt to help accomodate this.